packages/chat-component-bindings/src/hooks/TypeAssertions.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ErrorBar, MessageThread, ParticipantList, SendBox, TypingIndicator } from '@internal/react-components';
+import { ChatParticipantListSelector } from '../chatParticipantListSelector';
+import { ErrorBarSelector } from '../errorBarSelector';
+import { MessageThreadSelector } from '../messageThreadSelector';
+import { SendBoxSelector } from '../sendBoxSelector';
+import { TypingIndicatorSelector } from '../typingIndicatorSelector';
+import { getSelector } from './usePropsFor';
+import React from 'react';
+
+/**
+ * This function is a compile type check that {@link GetSelector} returns
+ * values of the correct type.
+ *
+ * @private
+ */
+export function assertGetSelectorTypes(): unknown {
```

Comment:
I love this one!!
Maybe we could type the return value a little bit more like
export function assertGetSelectorTypes(): [chatParticipantListProps, errorBarProps, ....] to see if props and selectors get perfect matched

Reply:
I think checking via return types and via local variable is identical ( see below ).

Using local variables keeps the type and the selector call close together.